packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -717,7 +719,7 @@ export class Function extends FunctionBase {
       kmsKeyArn: props.environmentEncryption?.keyArn,
       fileSystemConfigs,
       codeSigningConfigArn: props.codeSigningConfig?.codeSigningConfigArn,
-      architectures: architecture ? [architecture.name] : undefined,
+      architectures: this._architecture ? [this._architecture.name] : undefined,
```

Comment:
Ruh roh. Looks like we'll be able to have more than one?

Reply:
CloudFormation for some reason accepts a list with a max length of 1. We originally accepted a list, but then deprecated that in favor of a single value https://github.com/aws/aws-cdk/pull/16849.